contents/graham_scan/code/c++/graham_scan.cpp

```diff
@@ -0,0 +1,66 @@
+#include<iostream>
+#include<vector>
+#include<algorithm>
+
+struct point{
+  float x;
+  float y;
+};
+
+bool ccw(point a, point b, point c){
+  return ((b.x - a.x)*(c.y - a.y) >= (b.y - a.y)*(c.x - a.x));
+}
+
+std::vector<point> grahamScan(std::vector<point> points){
+  //selecting lowest point as pivot
+  int lowIndex = 0;
+  for (size_t i = 1; i < points.size(); i++) {
+    if(points[i].y < points[lowIndex].y){
+      lowIndex = i;
+    }
+  }
+  std::swap<point>(points[0], points[lowIndex]);
+  point pivot = points[0];
+
+  //sorting points by polar angle
+  sort(points.begin()+1, points.end(), [pivot](point a, point b){
+    return ccw(pivot, a, b);
+  });
+
+  //creating convex hull
+  size_t m = 1;
+  for (size_t i = 2; i <= points.size(); i++) {
+    while(ccw(points[m - 1], points[m], points[i]) <= 0){
+      if (m > 1) {
+                m--;
+                continue;
+            } else if (i == points.size()) {
+                break;
+            } else {
+                i++;
+            }
+    }
+    m++;
+    std::swap<point>(points[i],points[m]);
+  }
+  std::vector<point>::const_iterator first = points.begin();
+  std::vector<point>::const_iterator last = points.begin()+m;
+  std::vector<point> hull(first, last);
+  return hull;
+}
+
+void print(std::vector<point> points){
+  std::cout << "points in hull are as follows: \n";
+  for (size_t i = 0; i < points.size(); i++) {
```

Comment:
I prefer to use `for` *with range* similar with `for each` in other languages. Like this:
```suggestion
  for (auto p : points) {
```
Then use `p` instead of `points[i]` below.